OpenRA.Mods.Common/Traits/Buildings/RallyPoint.cs

```diff
@@ -73,22 +75,32 @@ public IEnumerable<IOrderTargeter> Orders
 		public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool queued)
 		{
 			if (order.OrderID == "SetRallyPoint")
-				return new Order(order.OrderID, self, false) { TargetLocation = self.World.Map.CellContaining(target.CenterPosition), SuppressVisualFeedback = true };
+				return new Order(order.OrderID, self, false) { TargetLocation = self.World.Map.CellContaining(target.CenterPosition), SuppressVisualFeedback = true,
+					ExtraData = ((RallyPointOrderTargeter)order).SetPrimaryBuilding ? ExtraDataSetPrimaryBuilding : 0 };
 
 			return null;
 		}
 
 		public void ResolveOrder(Actor self, Order order)
 		{
 			if (order.OrderString == "SetRallyPoint")
+			{
 				Location = order.TargetLocation;
+				if (order.ExtraData == ExtraDataSetPrimaryBuilding)
+				{
+					var primaryBuildingTrait = self.TraitOrDefault<PrimaryBuilding>();
```

Comment:
The benefit of this approach is that other mods that don't use `PrimaryBuilding` can attach their own special behaviour to the force-set command if they want to.  e.g. setting the rally point for all production structures.